### DISCUSSION TITLE
Calling 'finish' on a connecting LiveTranscription

### DISCUSSION THREAD

**Discussion by John-A-J** [deepgram/community#332] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

Using the javascript sdk "@deepgram/sdk": "2.3.0" we recently experienced a production error where LiveTranscription.finish() was called while still connecting.
```
Error: WebSocket is not open: readyState 0 (CONNECTING)
```
Our expected behaviour would be that the connection attempt is abandoned and no error is thrown.
We'll wrap the  in a try-catch for now

Please let me know if there is a better place to post this, thanks in advance

**Comment by jkroll-deepgram** [comment, marked as answer]
@John-A-J thanks for pointing out this behavior. Agreed that it's dicey needing to handle state to know that the websocket is open in order to close it. I looked into how Deepgram demonstrates handling this in our public examples. I recommend using our Node live example as a template. It nicely captures how you can handle the stream state. On your Deepgram client, you'll call `deepgram.getReadyState()` which yields either `1` (open), `2` (closing), or `3` (closed).
https://github.com/deepgram-devs/node-live-example/blob/main/server.js#L73-L86
